diff --git a/src/include/libplacebo/utils/libav_internal.h b/src/include/libplacebo/utils/libav_internal.h
index 4c269e5d..62996477 100644
--- a/src/include/libplacebo/utils/libav_internal.h
+++ b/src/include/libplacebo/utils/libav_internal.h
@@ -967,11 +967,6 @@ static void pl_fix_hwframe_sample_depth(struct pl_frame *out, const AVFrame *fra
     struct pl_bit_encoding *bits = &out->repr.bits;
 
     bits->sample_depth = fmt->component_depth[0];
-
-    switch (hwfc->sw_format) {
-    case AV_PIX_FMT_P010: bits->bit_shift = 6; break;
-    default: break;
-    }
 }
 
 static bool pl_map_avframe_drm(pl_gpu gpu, struct pl_frame *out,
@@ -1016,6 +1011,12 @@ static bool pl_map_avframe_drm(pl_gpu gpu, struct pl_frame *out,
     }
 
     pl_fix_hwframe_sample_depth(out, frame);
+	
+    switch (hwfc->sw_format) {
+    case AV_PIX_FMT_P010: out->repr.bits.bit_shift = 6; break;
+    default: break;
+    }
+
     return true;
 }
 
@@ -1157,7 +1158,6 @@ static bool pl_map_avframe_vulkan(pl_gpu gpu, struct pl_frame *out,
 
     out->acquire = pl_acquire_avframe;
     out->release = pl_release_avframe;
-    pl_fix_hwframe_sample_depth(out, frame);
     return true;
 }
 
diff --git a/src/vulkan/context.c b/src/vulkan/context.c
index ad8a859b..99879427 100644
--- a/src/vulkan/context.c
+++ b/src/vulkan/context.c
@@ -711,7 +711,8 @@ debug_layers_done: ;
         }
     }
 
-#ifdef VK_KHR_portability_enumeration
+    // only relevant for MacOS
+/*#ifdef VK_KHR_portability_enumeration
     // Required for macOS ( MoltenVK ) compatibility
     for (int n = 0; n < num_exts_avail; n++) {
         if (strcmp(VK_KHR_PORTABILITY_ENUMERATION_EXTENSION_NAME, exts_avail[n].extensionName) == 0) {
@@ -720,7 +721,7 @@ debug_layers_done: ;
             break;
         }
     }
-#endif
+#endif*/
 
     // Add extra user extensions
     for (int i = 0; i < params->num_extensions; i++) {
diff --git a/src/vulkan/swapchain.c b/src/vulkan/swapchain.c
index 0741fbfc..88bdf6a5 100644
--- a/src/vulkan/swapchain.c
+++ b/src/vulkan/swapchain.c
@@ -610,13 +610,21 @@ static bool vk_sw_recreate(pl_swapchain sw, int w, int h)
     }
 #endif
 
+    {
+        pl_gpu_flush(gpu);
+        vk_wait_idle(vk);
+        for (int i = 0; i < vk->pool_graphics->num_queues; i++)
+            vk->QueueWaitIdle(vk->pool_graphics->queues[i]);
+        vk->DestroySwapchainKHR(vk->dev, p->swapchain, PL_VK_ALLOC);
+    }
+
     // Calling `vkCreateSwapchainKHR` puts sinfo.oldSwapchain into a retired
     // state whether the call succeeds or not, so we always need to garbage
     // collect it afterwards - asynchronously as it may still be in use
-    sinfo.oldSwapchain = p->swapchain;
+    /* sinfo.oldSwapchain = p->swapchain; */
     p->swapchain = VK_NULL_HANDLE;
     VkResult res = vk->CreateSwapchainKHR(vk->dev, &sinfo, PL_VK_ALLOC, &p->swapchain);
-    vk_dev_callback(vk, (vk_cb) destroy_swapchain, vk, vk_wrap_handle(sinfo.oldSwapchain));
+    /* vk_dev_callback(vk, (vk_cb) destroy_swapchain, vk, vk_wrap_handle(sinfo.oldSwapchain)); */
     PL_VK_ASSERT(res, "vk->CreateSwapchainKHR(...)");
 
     // Get the new swapchain images
